src/core/thread/thread_netif.cpp

```diff
@@ -171,6 +172,23 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
+ThreadError ThreadNetif::TmfFilter(const otMessage *aMessage, const otMessageInfo *aMessageInfo)
+{
+    const Ip6::MessageInfo &messageInfo = *static_cast<const Ip6::MessageInfo *>(aMessageInfo);
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(messageInfo.GetSockAddr().IsRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsRoutingLocator() ||
+                 messageInfo.GetSockAddr().IsLinkLocal() ||
+                 messageInfo.GetSockAddr().IsAnycastRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsAnycastRoutingLocator(),
```

Comment:
Sorry if my previous comment confused you. 
What I want to say is ALOC(`IID: 0000:00FF:FE00:FCxx`) shares the similar address structure as RLOC(`IID:0000:00FF:FE00:xxxx`). For an ALOC address, `IsRoutingLocator()` should always be true, thus no need for the condition` IsAnycastRoutingLocator() `.